File: client/src/app/_models/userparams.ts
import { User } from "./user";

export class UserParams {
  gender: string;
  minAge = 18;
  maxAge = 1000;
  pageNumber = 1;
  pageSize = 5;
  orderBy: string = 'lastActive';


  constructor(user: User | null) {
    this.gender = user?.gender === 'female' ? 'male' : 'female';
  }
}

File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string? Gender { get; set; }
        public string? CurrentUsername { get; set; }

        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 1000;
        public string orderBy { get; set; } = "lastActive";

    }
}